### PR TITLE
Apache Pulsar: Add support for partitioned topics

### PR DESCRIPTION
**Pull request by michaeljmarshall**
Signed-off-by: Michael Marshall <mmarshall@apache.org>

Apache Pulsar has topics and partitioned topics. In order to support scaling partitioned topics, the Scaler needs to call a different stats endpoint. The data returned by that endpoint is in the same format as the one returned by the non-partitioned stats endpoint.

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes https://github.com/kedacore/keda/issues/3833
Relates https://github.com/kedacore/keda-docs/pull/975

### CODE REVIEW

**Comment by zroubalik** [comment]
@michaeljmarshall looking good, could you please rebase the branch?

File: pkg/scalers/pulsar_scaler.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type pulsarScaler struct {
 type pulsarMetadata struct {
 	adminURL                      string
 	topic                         string
+	isPartitionedTopic            bool
```
Comment (zroubalik) [inline pkg/scalers/pulsar_scaler.go:31]: this property is not read anywhere, what is the purpose then?

Comment (michaeljmarshall) [reply pkg/scalers/pulsar_scaler.go:31]: I misunderstood the data model. I'll remove this field since it is not necessary.

**Comment by zroubalik** [comment]
@michaeljmarshall e2e test failed.

**Comment by michaeljmarshall** [comment]
Thanks @zroubalik. I tested locally and think these changes should work now.

**Review by eolivelli — APPROVED** [review]
LGTM
nice work!

**Review by zroubalik — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**